README.md

```diff
@@ -1,26 +1,178 @@
 
-Get in touch [here](https://vimeo.com/help/contact).
+```Swift
+let appConfiguration = AppConfiguration(
+		clientIdentifier: "Your client identifier goes here",
+		clientSecret: "Your client secret goes here",
+		scope: [.Public, .Private, .Interact])
+```
+
+### Client
+
+To interact with the Vimeo API, we use a `VimeoClient` instance.  The client is the primary interface of **VimeoNetworking**, it takes care of executing requests, cache storage and retrieval, tracking the current authenticated account, and handling global error conditions.  You can instantiate one with the application configuration you created a moment ago.
+
+```Swift
+let vimeoClient = VimeoClient(configuration: appConfiguration)
+```
+
+## Authenticating
+
+Before we can actually start getting meaningful data from the API, there's one last step: authentication. `AuthenticationController` handles and simplifies this process.  It uses the configuration of an associated `VimeoClient` to make requests. On successful authentication, it passes the new Vimeo account back to this client, and that client can then make requests to API endpoints.
+
+### Client Credentials
+
+Client credentials allow you to see everything that's publicly available on Vimeo.  This is essentially equivalent to visiting Vimeo.com without signing up for an account or logging in.  This is the simplest authentication method to implement, just one function completes the grant.
+
+```Swift 
+let authenticationController = AuthenticationController(client: vimeoClient)
+
+authenticationController.clientCredentialsGrant { result in 
+	switch result {
+	case .Success(let account):
+		print("Successfully authenticated with account: \(account)")
+	case .Failure(let error):
+		print("error authenticating: \(error)")	
+	}
+}
+```
+
+### Code Grant
+
+If you want to log in as a user, the Vimeo API provides a process called Code Grant authentication.  This is a bit more involved than a client credentials grant, but it gives your application the benefit of making requests on behalf of a Vimeo user, and viewing their personal content.  
+
+To authenticate via code grant, your app launches a specific URL in Safari.  The user signs up or logs in on Vimeo.com, and chooses which permissions to grant.  Then, control is redirected back to your application where authentication completes.
+
+To prepare your application to receive this redirect, navigate to your app target settings > Info > URL Types.  Add a new URL Type, and under url scheme enter `vimeo` followed by your client identifier (for example, if your client identifier is `1234`, enter `vimeo1234`).  You also need to add this redirect URL to your app on the Vimeo API site.  Under “App Callback URL”, add `vimeo{CLIENT_KEY}://auth` (for the example above, `vimeo1234://auth`).
```

Comment:
Should there be a possessive here? 

"navigate to your app's target settings"
"navigate to your app target's settings"